recipes-bsp/imx-mkimage/imx-boot_1.0.bb

```diff
@@ -84,32 +84,33 @@ compile_mx8m() {
 
     cp ${DEPLOY_DIR_IMAGE}/signed_dp_imx8m.bin               ${BOOT_STAGING}
     cp ${DEPLOY_DIR_IMAGE}/signed_hdmi_imx8m.bin             ${BOOT_STAGING}
-    cp ${DEPLOY_DIR_IMAGE}/u-boot-spl.bin-${MACHINE}-${UBOOT_CONFIG_EXTRA} \
+    cp ${DEPLOY_DIR_IMAGE}/u-boot-spl.bin-${MACHINE}${UBOOT_CONFIG_EXTRA_FN} \
                                                              ${BOOT_STAGING}/u-boot-spl.bin
 
     if [ "x${UBOOT_SIGN_ENABLE}" = "x1" ] ; then
         # Use DTB binary patched with signature node
-        cp ${DEPLOY_DIR_IMAGE}/${UBOOT_DTB_BINARY} ${BOOT_STAGING}/${UBOOT_DTB_NAME_EXTRA}
+        cp ${DEPLOY_DIR_IMAGE}/${UBOOT_DTB_BINARY}           ${BOOT_STAGING}/${UBOOT_DTB_NAME_EXTRA}
     else
-        cp ${DEPLOY_DIR_IMAGE}/${BOOT_TOOLS}/${UBOOT_DTB_NAME_EXTRA}   ${BOOT_STAGING}
+        cp ${DEPLOY_DIR_IMAGE}/${BOOT_TOOLS}/${UBOOT_DTB_NAME}   ${BOOT_STAGING}
     fi
 
-    cp ${DEPLOY_DIR_IMAGE}/${BOOT_TOOLS}/u-boot-nodtb.bin-${MACHINE}-${UBOOT_CONFIG_EXTRA} \
+    if [ -e ${DEPLOY_DIR_IMAGE}/${BOOT_TOOLS}/u-boot-nodtb.bin-${MACHINE}${UBOOT_CONFIG_EXTRA_FN} ]; then
+        cp ${DEPLOY_DIR_IMAGE}/${BOOT_TOOLS}/u-boot-nodtb.bin-${MACHINE}${UBOOT_CONFIG_EXTRA_FN} \
                                                              ${BOOT_STAGING}/u-boot-nodtb.bin
+    fi
 
     cp ${DEPLOY_DIR_IMAGE}/${ATF_MACHINE_NAME}               ${BOOT_STAGING}/bl31.bin
-
-    cp ${DEPLOY_DIR_IMAGE}/${UBOOT_NAME_EXTRA}                     ${BOOT_STAGING}/u-boot.bin
-
+    cp ${DEPLOY_DIR_IMAGE}/${UBOOT_NAME_EXTRA}               ${BOOT_STAGING}/u-boot.bin
 }
+
 compile_mx8() {
     bbnote 8QM boot binary build
     cp ${DEPLOY_DIR_IMAGE}/${BOOT_TOOLS}/${SC_FIRMWARE_NAME} ${BOOT_STAGING}/scfw_tcm.bin
     cp ${DEPLOY_DIR_IMAGE}/${ATF_MACHINE_NAME}               ${BOOT_STAGING}/bl31.bin
```

Comment:
Hi @angolini, I happened to be looking at taking this change into our local layer, but I'm confused how it even works. Since this function can be called multiple times, it will fail on the second call for any files which are not UBOOT_CONFIG specific (like the ATF file here) since in that case it will be copying the same file to the same location.

Reply:
I agree that we have to think of a better way to reorg this recipe. It's becoming more and more disorganized and crazy. But maybe it would be better if we only switch to bootloader container instead?

If you have any suggestions on how to fix/enhance the PR, please let me know.

Reply:
Yes, I believe moving to the bootloader container would be better, but in the meantime, I guess the recipe should at least be fixed.

A refactoring of the functions is needed so common logic occurs once and uboot-config-specific logic occurs once per config.

Reply:
I changed where the spl is copied.

@thochstein , if you know a better way to test this (without the machines, as I don't have them) please, let me know

Reply:
Can you just put two values in `UBOOT_CONFIG`? I'm pretty sure that will just fail to build.

Reply:
```
./tmp/sysroots-components/imx8mm_lpddr4_evk/imx-boot/boot/imx-boot-imx8mm-lpddr4-evk-sd.bin-flash_evk_flexspi
./tmp/sysroots-components/imx8mm_lpddr4_evk/imx-boot/boot/imx-boot-imx8mm-lpddr4-evk-fspi.bin-flash_evk_flexspi

```

I've been testing this as well.

But if you think in anything else, please, let me know

(this PR does not touch the multiple UBOOT_CONFIG, it only deals when UBOOT_CONFIG is empty)